### PR TITLE
Add configuration history routines.

### PR DESCRIPTION
**Pull request by mcb64**
Added the ability to pull out a configuration history between specific times.

This was added as a command line option to pmgrUtils, with the option of producing a short list of changes on the terminal or save the complete set of changed configs into a CSV file.

### CODE REVIEW

File: pmgr/pmgrAPI.py [hunk 1]
```diff
@@ -339,3 +339,56 @@ def add_hutch(self, newhutch):
...
+
+    @staticmethod
+    def _get_datetime(s):
+        """
+        Try to convert a string into a datetime object.  The following
+        conversions are attempted:
+             ISO format
+             mm/dd/yyyy
+             mm/dd/yyyy hh:mm
+        If all three fail, a ValueError exception is raised.
```
Comment (ZLLentz) [inline pmgr/pmgrAPI.py:351]: This docstring line isn't correct based on the as-implemented code

File: pmgr/pmgrobj.py [hunk 2]
```diff
@@ -778,6 +778,59 @@ def configChange(self, idx, e, update=True):
...
+            elif finish is not None:
+                    cmd = cmd + " and dt_updated <= '%s'" % finish
+            self.cur.execute(cmd)
+            l = []
+            p = None
+            for r in self.cur.fetchall():
+                l.append(self._cleanHistory(r, p, diff))
+                p = r
+            return l
+        except:
```
Comment (ZLLentz) [inline pmgr/pmgrobj.py:831]: Note that bare excepts also catch e.g. `KeyboardInterrupt`
```suggestion
        except Exception:
```

**Review by ZLLentz — COMMENTED** [review]
Looks useful, I'll also fetch the commit and try it myself

File: pmgr/pmgrAPI.py (hunk 1 above)
Comment (mcb64) [reply pmgr/pmgrAPI.py:351]: Agreed.

File: pmgr/pmgrobj.py (hunk 2 above)
Comment (mcb64) [reply pmgr/pmgrobj.py]: Agreed.

File: pmgr/pmgrUtils.py [hunk 3]
```diff
@@ -30,6 +31,9 @@
     apply          Apply the saved motor configuration to live values
     diff           Prints differences between pmgr and live values
     find           Find the configurations matching the given pattern
+    history        Give the history of the configuration between the start
+                   and end dates.  Save a full listing as a CSV file,
+                   or get a difference between successive changes.
```
[the remark this answers is not in the document]
Comment (mcb64) [reply pmgr/pmgrUtils.py]: Actually, there are three possible formats, so I'll give an example of each.

**Review by ZLLentz — APPROVED** [review]
Looks good to me!

**Pull request merged**